app/interactions/user_submits_a_set.rb

```diff
@@ -16,6 +16,7 @@ def call(set)
       else
         lobby.decrement_score_for(user)
       end
+      check_game_state
```

Comment:
this is a rather ambiguous name for a method (probably because the method does two unrelated things), and the notion of `checking` doesn't really represent the fact this can alter the game.